File: factory_pattern/java/pizza_store/pizza_factory_method/PizzaTestDrive.java
package pizza_store.pizza_factory_method;


public class PizzaTestDrive {

    public static void main(String[] args) {
        PizzaStore NewYorkStore = new NewYorkPizzaStore();
        PizzaStore ChicagoStore = new ChicagoPizzaStore();

        Pizza pizza = NewYorkStore.orderPizza("cheese");
        System.out.println("Ethan ordered a " + pizza.getName() + "\n");

        pizza = ChicagoStore.orderPizza("pepperoni");
        System.out.println("Joel ordered a " + pizza.getName() + "\n");
    }
}

File: factory_pattern/python/pizza_store/pizza_test_drive.py
from pizza_factory_method import NewYorkPizzaStore, ChicagoPizzaStore


def main():
    new_york_store = NewYorkPizzaStore()
    chicago_store = ChicagoPizzaStore()

    pizza = new_york_store.order_pizza("cheese")
    print(f"Ethan ordered a {pizza.name}\n")

    pizza = chicago_store.order_pizza("pepperoni")
    print(f"Joel ordered a {pizza.name}\n")


if __name__ == '__main__':
    main()